src/bitcoin-cli.cpp

```diff
@@ -867,6 +868,71 @@ static void GetWalletBalances(UniValue& result)
     result.pushKV("balances", balances);
 }
 
+static void ParseGetInfoResult(UniValue& result)
+{
+    if (!find_value(result, "error").isNull()) {
+        return;
+    }
+
+#ifndef WIN32
+    const std::string RESET = "\x1B[0m";
+    const std::string GREEN = "\x1B[32m";
+    const std::string BLUE = "\x1B[34m";
+    const std::string YELLOW = "\x1B[33m";
+    const std::string MAGENTA = "\x1B[35m";
+    const std::string CYAN = "\x1B[36m";
+#else
+    const std::string RESET = "";
+    const std::string GREEN = "";
+    const std::string BLUE = "";
+    const std::string YELLOW = "";
+    const std::string MAGENTA = "";
+    const std::string CYAN = "";
+#endif
+
+    std::string result_string = strprintf("%sNetwork%s\n", GREEN, RESET);
+    result_string += strprintf("version: %s\n", result["version"].getValStr());
+    result_string += strprintf("time offset: %s\n", result["timeoffset"].getValStr());
+    result_string += strprintf("proxy: %s \n", result["proxy"].getValStr());
+    result_string += strprintf("connections: in %s, out %s, total %s\n",
+                               result["connections"]["in"].getValStr(),
+                               result["connections"]["out"].getValStr(),
+                               result["connections"]["total"].getValStr());
+    result_string += strprintf("relay fee: %s\n\n", result["relayfee"].getValStr());
+
+    result_string += strprintf("%sBlock chain%s\n", BLUE, RESET);
+    result_string += strprintf("blocks: %s\n", result["blocks"].getValStr());
+    result_string += strprintf("headers: %s\n", result["headers"].getValStr());
+    result_string += strprintf("verification progress: %s\n", result["verificationprogress"].getValStr());
+    result_string += strprintf("difficulty: %s\n", result["difficulty"].getValStr());
+    result_string += strprintf("chain: %s\n\n", result["chain"].getValStr());
+
+    if (!result["has_wallet"].isNull()) {
+        result_string += strprintf("%sWallet%s\n", MAGENTA, RESET);
+        result_string += strprintf("keypool size: %s\n", result["keypoolsize"].getValStr());
+        if (!result["unlocked_until"].isNull()) {
+            result_string += strprintf("unlocked_until: %s\n", result["unlocked_until"].getValStr());
+        }
+        result_string += strprintf("paytxfee: %s\n\n", result["paytxfee"].getValStr());
+    }
+    if (!result["balance"].isNull()) {
+        result_string += strprintf("%sBalance%s: \u20BF%s\n\n", CYAN, RESET, result["balance"].getValStr());
+    }
+
+    if (!result["balances"].isNull()) {
+        result_string += strprintf("%sBalances%s\n", CYAN, RESET);
+
+        for (std::string wallet : result["balances"].getKeys()) {
```

Comment:
Can reference to const here rather than copying
 ```suggestion
        for (const std::string& wallet : result["balances"].getKeys()) {
```

Reply:
Updated to use string reference instead of copying